CUDADataFormats/Track/interface/TrackSoAHeterogeneousT.h

```diff
@@ -19,39 +19,61 @@ class TrackSoAHeterogeneousT {
   using hindex_type = uint32_t;
   using HitContainer = cms::cuda::OneToManyAssoc<hindex_type, S, 5 * S>;
 
+
+  // quality accessors
+  constexpr Quality quality(int32_t i) const { return reinterpret_cast<Quality>(quality_(i)); }
+  constexpr Quality &quality(int32_t i) { return reinterpret_cast<Quality &>(quality_(i)); }
+  constexpr Quality const *qualityData() const { return reinterpret_cast<Quality const *>(quality_.data()); }
+  constexpr Quality *qualityData() { return reinterpret_cast<Quality *>(quality_.data()); }
+
+  // chi2 accessors
+  __host__ __device__ auto & chi2(int32_t i) { return chi2_(i); }
+  __host__ __device__ auto const chi2(int32_t i) const { return chi2_(i); }
```

Comment:
if these are returned by value, there is no point/need for the `const` qualifier in the return type

Reply:
Indeed, I will fix that.